### PR TITLE
docs: add guide for unit testing

### LINKED ISSUE #27754: [Feature] Unit Testing
I have read about experimental component testing, but it would be great if we could use playwright also as a unit testing framework for singular functions, so no jest or vitest ist needed alongside. Is this currently already possible ?

### PR DESCRIPTION
**Pull request by boan-anbo**
This closes #27754

### CODE REVIEW

**Comment by MindaugasMateika** [comment]
In my opinion, a lot of unit testing usually involves mocking/stubbing of some sort. Playwright does not have this capability but I think it would be beneficial if not adding an example then just mentioning that mocking/stubbing should be performed with some 3rd  party libraries.

**Comment by pavelfeldman** [comment]
Is this something you are actively working on? (docs / lint did not pass)

**Comment by boan-anbo** [comment]
> Is this something you are actively working on? (docs / lint did not pass)

yes, I was waiting for early feedback. If the lint needs to pass first to be reviewed, just let me know!

**Comment by boan-anbo** [comment]
@microsoft-github-policy-service agree

File: docs/src/test-unit-testing-js.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+});
+
+test(`testing with custom fixture`, async ({ defaultItem }) => {
+  expect(defaultItem).toBe("defaultItemName");
+});
+```
+
+## Required browser configuration
+
+Note that even for unit tests without browsers, Playwright requires at least one browser configuration in your [`config`](./browsers#configure-browsers) to run any tests at all. Without it, no tests will be picked up for execution, regardless of whether they use browser-related APIs or not.
```
Comment (yury-s) [inline docs/src/test-unit-testing-js.md:89]: This is false. The project section could be omitted altogether, or one can have a project without `use` section.

**Review by yury-s — COMMENTED** [review]
The idea we want to convey is that Playwright will only create fixtures that are used by the tests. If the tests don't use browser/context/page fixtures, no browsers are launched and none of the fixtures are created. In that case the user also doesn't need to install the browsers. Reading the document it's not clear to me. I'd rather we focused on this aspect and drop the parts where we explain differences between unit and e2e tests.

**Comment by dgozman** [comment]
This one seems stale, closing.

**Pull request closed without merging**